services/storage/src/simcore_service_storage/cli.py

```diff
@@ -1,120 +1,97 @@
-    ``simcore_service_storage.__main__`` in ``sys.modules``.
-  - When you import __main__ it will get executed again (as a module) because
-    there's no ``simcore_service_storage.__main__`` in ``sys.modules``.
-
-"""
-import json
 import logging
 import os
-import sys
-from pathlib import Path
 from pprint import pformat
-from typing import Dict, List, Optional, Union
 
-import click
-import yaml
+import typer
 from pydantic import ValidationError
+from pydantic.env_settings import BaseSettings
 
 from . import application
-from .resources import resources
-from .settings import ApplicationSettings
+from .settings import Settings
+
+HEADER = "{:-^50}"
+
 
 log = logging.getLogger(__name__)
+main = typer.Typer(name="osparc-simcore storage service")
+
+
+def print_as_envfile(settings_obj, *, compact, verbose):
+    for name in settings_obj.__fields__:
+        value = getattr(settings_obj, name)
+
+        if isinstance(value, BaseSettings):
+            if compact:
+                value = f"'{value.json()}'"  # flat
+            else:
+                if verbose:
+                    typer.echo(f"\n# --- {name} --- ")
+                print_as_envfile(value, compact=False, verbose=verbose)
+                continue
 
-JsonNode = Union[Dict, List]
-
-
-def prune_config(cfg: JsonNode):
-    """
-    Converts
-    {
-        max_workers: 8
-        monitoring_enabled: ${STORAGE_MONITORING_ENABLED}
-        test_datcore:
-            token_key: ${BF_API_KEY}
-            token_secret: ${BF_API_SECRET}
-    }
-        -->
-
-    {
-        max_workers: 8
-        test_datcore: {}
-    }
-
-    """
-    if isinstance(cfg, Dict):
-        for key in list(cfg.keys()):
-            value = cfg[key]
-            if isinstance(value, str) and value.startswith("${"):
-                del cfg[key]
-            elif isinstance(value, (List, Dict)):
-                prune_config(cfg[key])
-
-    elif isinstance(cfg, List):
-        for item in cfg:
-            prune_config(item)
-
-
-@click.command("Service to manage data storage in simcore.")
-@click.option(
-    "--config",
-    default=None,
-    type=click.Path(exists=False, path_type=str),
-)
-@click.option(
-    "--check-config",
-    "-C",
-    default=False,
-    is_flag=True,
-    help="Checks settings, prints and exit",
-)
-def main(config: Optional[Path] = None, check_config: bool = False):
-    config_dict = {}
+        if verbose:
+            field_info = settings_obj.__fields__[name].field_info
+            if field_info.description:
+                typer.echo(f"# {field_info.description}")
+
+        typer.echo(f"{name}={value}")
+
+
+def print_as_json(settings_obj, *, compact=False):
+    typer.echo(settings_obj.json(indent=None if compact else 2))
+
+
+@main.command()
+def settings(
+    as_json: bool = False,
+    as_json_schema: bool = False,
+    compact: bool = typer.Option(False, help="Print compact form"),
+    verbose: bool = False,
+):
```

Comment:
For each service we have to define this? Would it not be more convent to have it in some common place? Or did I misunderstand the usage of this command. I thought it should be used to list the settings of a service.

Reply:
Good idea. I will move it to ``servicelib``

Reply:
I have tried but noticed that i need to add the ``typer`` dependency . I would like to avoid that until i have both the webserver and the storage refactored (see notes on next PRs above). Then I will implement the CLIs of all the services using ``typer`` and add to ``servicelib`` the common commands.